geopy/geocoders/herev7.py

```diff
@@ -0,0 +1,286 @@
+import collections.abc
+import warnings
+from functools import partial
+from urllib.parse import urlencode
+
+from geopy.exc import (
+    ConfigurationError,
+    GeocoderAuthenticationFailure,
+    GeocoderInsufficientPrivileges,
+    GeocoderQuotaExceeded,
+    GeocoderServiceError,
+    GeocoderUnavailable,
+)
+from geopy.geocoders.base import DEFAULT_SENTINEL, Geocoder
+from geopy.location import Location
+from geopy.util import join_filter, logger
+
+__all__ = ("HereV7", )
+
+
+class HereV7(Geocoder):
+    """Geocoder using the HERE Geocoder API.
+
+    Documentation at:
+        https://developer.here.com/documentation/geocoding-search-api/
+    """
+
+    structured_query_params = {
+        'street',
+        'houseNumber',
+        'postalCode',
+        'city',
+        'district',
+        'county',
+        'state',
+        'country'
+    }
+
+    geocode_path = '/v1/geocode'
+    reverse_path = '/v1/revgeocode'
+
+    def __init__(
+            self,
+            *,
+            apiKey=None,
+            scheme=None,
+            timeout=DEFAULT_SENTINEL,
+            proxies=DEFAULT_SENTINEL,
+            user_agent=None,
+            ssl_context=DEFAULT_SENTINEL,
+            adapter_factory=None
+    ):
+        """
+
+        :param str apiKey: Should be a valid HERE Maps apiKey.
+            More authentication details are available at
+            https://developer.here.com/authenticationpage.
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :param str user_agent:
+            See :attr:`geopy.geocoders.options.default_user_agent`.
+
+        :type ssl_context: :class:`ssl.SSLContext`
+        :param ssl_context:
+            See :attr:`geopy.geocoders.options.default_ssl_context`.
+
+        :param callable adapter_factory:
+            See :attr:`geopy.geocoders.options.default_adapter_factory`.
+
+            .. versionadded:: 2.0
+        """
+        super().__init__(
+            scheme=scheme,
+            timeout=timeout,
+            proxies=proxies,
+            user_agent=user_agent,
+            ssl_context=ssl_context,
+            adapter_factory=adapter_factory,
+        )
+
+        domain = "search.hereapi.com"
+
+        self.apiKey = apiKey
+        self.api = "%s://geocode.%s%s" % (self.scheme, domain, self.geocode_path)
+        self.reverse_api = (
+            "%s://revgeocode.%s%s" % (self.scheme, domain, self.reverse_path)
+        )
+
+    def geocode(
+        self,
+        query,
+        components=None,
```

Comment:
`components` should be behind the asterisk as well, because the base `geocoder` method doesn't have that arg.

Reply:
Fixed in 3072157